### DISCUSSION TITLE
G-code preview different from STL

### DISCUSSION THREAD

**Discussion by UD551** [supermerill/SuperSlicer#1411] [Q&A] [answered]
Hello, I am using the newest version of Super Slicer and I am facing the following problem that I have not been able to fix. 
I am trying to print closed channels (like a tunnel) inside a cube. However, the G-code preview is resulting in a different outcome than the STL. For example, I have a square Channel of (200x200), which means the height of the channel should be 200. However, the G-code is not accounting for that and is closing the top of the channel at a height of 250 which is one layer higher than the assigned STL.
In other words, as shown in the GIF below, the STL file shows that the top of the Channel must close  at a layer height of 0.35, but what is actually happening, it is closing at a layer of 0.4
The only right thing happening is the perimeter closing the top edge of the opening at the correct layer of 0.35
What should I do to make the Internal infill close the top of the channel at the correct layer height of 0.35? Is there any setting I could play with?
Thank you in advance for your time. I appreciate any kind of help.
  
![Bottle Neck GIf](https://user-images.githubusercontent.com/87382555/125494727-c2be76d9-c6e9-4253-ace8-772f342b2c28.gif)

**Comment by neophyl** [comment]
With these sorts of things its always best to save your project from Super Slicer (File>Save project as) then take the resulting .3mf file, zip it up into a zip archive and attach it.  Must be zip file or github wont allow it.  A project file saved from SuSi will contain not only the model but all of your settings as well as any modifiers or anything else in play.  Without it is very hard to debug issues as its all guess work.

If I had to guess with the information you have provided I would ask what layer height you are using.  With a 0.2mm layer height then it cant print at 0.35 can it.  Only at 0.2/0.4/0.6 etc.  Also from your picture it looks like you have something very strange going on with your extrusion widths as the layers directly above the channel are printed much much wider.

Please post your 3mf project file.

**Comment by UD551** [comment]
Hello, Thank you for responding. I really appreciate it. 
I will try to re-explain my problem more clearly.  
I am trying to build a channel inside a filled cube. I am facing an issue with the top of the channel not being filled at the correct height.
In this project, the channel is a square with dimensions of 150x150 microns. I have set it to a layer height of 50 microns, so that the channel would be made up of 3 layers. However, when I preview the G-code in SuSi it shows that the channel top (internal infill) is being printed at a layer height of 0.4 rather than 0.35, which indicates that the channel height is turning out to be 200microns rather than 150microns.

I tried an approach using height range modifiers to change the width of internal infill at layer of z= 0.35 (top of the channel) in order to try and fill out the channel top at the correct z height. However, this approach didn't. 
As a side note, you can see that the perimeter is the only thing being printed correctly at a layer of z=0.35. I ideally want the top of the channel to also fill out at a layer of z = 0.35 rather than filling out at z = 0.4. 

Thank you in advance for the help. I hope I was able to portray my problem clearly. 


[Channel_150_Final.zip](https://github.com/supermerill/SuperSlicer/files/6816720/Channel_150_Final.zip)

**Comment by neophyl** [comment, marked as answer]
Thank you for providing the 3mf.  Not sure why its doing that as it really shouldnt.  I did find a work around though.
[Channel_150_Final_neo.zip](https://github.com/supermerill/SuperSlicer/files/6817674/Channel_150_Final_neo.zip)

Please see the attached project file.  I managed to work around it by removing any bottom layers.  I've also changed a few other settings as your 0.8 widths wont fill in nicely internally and by putting the layer height back to 0.15 (the height of your channel) and then using a height range modifier on the top to put it down to 0.05 it can be made to print faster while still being the required 0.65 height.  

I also took the opportunity to repair your model in case it was contributing to weird slicing issues as it had several non manifold edges and overlapping faces. You had internal faces where there should have been and lots of other little issues. It wasn't the cause but its best to rule out model geometry.  

Hopefully the project will be of use.

**Reply by UD551** [reply]
Thank you so much for the help! I will be trying it and getting back to you <3.
